### ISSUE TITLE
glazedlists combox doesn't handle "\n" compatibly

### ISSUE THREAD

**Issue by errael** [glazedlists/glazedlists#695]
To reproduce with `AutoCompleteSupportTestApp`, add a newline into `URL_SAMPLE_DATA `, for example between the "oo"
```
        "http://mail.go\nogle.com/mail/",
```

Observe that glazed draws two lines, the normal JComboBox, draws one line, replacing the "\n" with " ".

![GlazedComboNewLine](https://user-images.githubusercontent.com/20450427/105130795-86ecdc80-5a9c-11eb-8020-49b6a1ec7530.png)

The problem is that the document property `"filterNewlines"` is being ignored by glazed's filterBypass. See `PlainDocument.insertString()`. This property is set by `JTextField.setDocument(doc)`

**Comment by bpangburn** [comment]
Resolving this would make our app look a lot nicer as we sometimes pickup items for our lists that can have line breaks.

**Comment by hbrands** [comment]
I'm not sure that this is Glazed Lists' fault.
The `filterBypass `object is not created by Glazed Lists, it's the default one. This does not handle the property at all.

If you set a document filter on the regular ComboBox in `AutoCompleteSupportTestApp` like this:

```
        JTextField textField = (JTextField) regularComboBox.getEditor().getEditorComponent();
        ((AbstractDocument) textField.getDocument()).setDocumentFilter(new DocumentFilter());
```
then the display of the item is also not correct in the editor.

**Comment by errael** [comment]
> as soon as you use _**any**_ DocumentFilter with JComboBox, the document property "filterNewlines" seems to be ignored.

That's true. And it's also easy to interpret the javadoc that _**any**_ mutations/filtering of the input string becomes the responsibility of the `DocumentFilter`. I'm not trying to be a lawyer, I'm just looking to get the behavior fixed.

I've got a patch in the works. Is there any interest in it?
